src/manifolds/Sphere.jl

```diff
@@ -412,12 +413,12 @@ function Base.show(io::IO, ::ArraySphere{N,𝔽}) where {N,𝔽}
 end
 
 """
-    uniform_distribution(M::AbstractSphere, p)
+    uniform_distribution(M::Sphere{n,ℝ}, p) where {n}
 
-Uniform distribution on given [`AbstractSphere`](@ref) `M`. Generated points will be of
+Uniform distribution on given [`Sphere`](@ref) `M`. Generated points will be of
 similar type as `p`.
 """
-function uniform_distribution(M::AbstractSphere, p)
+function uniform_distribution(M::Sphere{n,ℝ}, p) where {n}
```

Comment:
I had thought with the projection, this should also work for the `AbstractSphere`; why was this reduced to the sphere?

Reply:
Distributions is fairly aggressive with checking types in the constructor, so `MvNormal` will error if passed any `p` that is not a real vector. So I guess could I could make the type `M::Union{Sphere{n,ℝ},ArraySphere{Tuple{n},ℝ}}`.

One way to extend to the full `AbstractSphere` would be to have a function that represents `p` as a real vector by flattening and interleaving (e.g. complex numbers become 2 real numbers), then replacing `project!` with a function that first un-flattens then projects. But that seems outside the scope of this PR.

Reply:
Ah, so theoretically it would work but `MvNormal` is a little too strictly typed. No let's not make it too complicated, it can stay this way then.